std/json.c3

```diff
@@ -0,0 +1,428 @@
+@require_once "std/arithmetic.c3"
+@require_once "std/string.c3"
+@require_once "std/memory.c3"
+@require_once "std/vector.c3"
+@require_once "std/wrappers.c3"
+@require_once "std/iterators.c3"
+
+typedef Cursor : {
+    data      : StringView,
+    position  : isize, // TODO should this be an iterator?
+    allocator : Allocator,  // FIXME use global allocator.
+}
+
+typedef JSON_Node : {
+    type     : int, // 0: null, 1: true, 2: false, 3: string, 4: number,
+                    // 5: array, 6: object
+    value    : Optional<StringView>,
+    children : Optional<Vector<JSON_Node>>,
+}
+
+// TODO something like new<JSON_Node> would be nice, but we don't support
+// function specializations.
+function new_json_node : (node_type : int) -> JSON_Node = {
+    return {
+        node_type,
+        make_optional<StringView>(),
+        make_optional<Vector<JSON_Node>>(),
+    };
+}
+
+function char : (literal : u8[1]&) -> u8 = {
+    return literal[0];
+}
```

Comment:
Ooo, please add this to the standard library. It would also make the parser much cleaner

Reply:
Sure, although I still want character literals.